elastic/kibana
# [Fleet] Allow user-configured namespace for managed policies

## Summary

Resolves #110533. This PR addresses two concerns for preconfigured managed agent policies:

- Although we allow the namespace field to be edit in the UI, the user-configured value is overridden with the original value in `kibana.yml` whenever Fleet setup is triggered. This PR fixes it so that namespace is not overridden any more.
- We also allow the agent monitoring settings to be edited in the UI (checkboxes for logs and metrics collection), user changes are also overridden in a similar way. In this case we do not want these settings to be editable by the user for managed policies, so this PR disables the checkboxes in the UI.

![image](https://user-images.githubusercontent.com/1965714/139160298-c33b0bca-bdfb-4c24-9e71-29f4d9428624.png)

## How to test

1. Set up a preconfigured managed policy in `kibana.yml` for example:
```
xpack.fleet.packages:
  - name: system
    version: latest

xpack.fleet.agentPolicies:
  - name: Test policy
    description: Testing managed preconfiguration
    is_default: true
    is_managed: true
    namespace: default
    monitoring_enabled: []
    package_policies:
      - name: system-1
        package:
          name: system
```

2. Go to the agent policy details > Settings tab. The agent monitoring options should be disabled.
3. Edit the namespace field, save it.
4. Go back to Fleet home page, do a hard refresh to trigger Fleet setup.
5. Go to the agent policy details > Settings tab. The namespace field should be the value you had entered previously.
